Bugfix +android x conversion + version updates

MFlisar opened the pull request:
I did following:

* fixed round button text position
* versions updated
* android x conversion done
* extracted definitions to versions.gradle for easier reusage in own projects

If you don't want to merge this because you don't want to migrate to android x, you should at least check out this change I made, as it solves a bug in your library (if rectangular highlights are used, the text may have been moved off screen):

https://github.com/MFlisar/FancyShowCaseView/blob/235b4befc0862efb34ac3a46d047fe7641ce8162/fancyshowcaseview/src/main/java/me/toptas/fancyshowcase/Calculator.kt#L149-L159

faruktoptas commented on fancyshowcaseview/src/main/java/me/toptas/fancyshowcase/Calculator.kt:
```diff
@@ -146,12 +146,14 @@ class Calculator(activity: Activity,
         val spaceBelow = bitmapHeight - bottom.toInt()
         val params = view.layoutParams as RelativeLayout.LayoutParams
 
+        val viewHeight = if (focusShape == FocusShape.ROUNDED_RECTANGLE) focusHeight else viewRadius
```
It should be `focusHeight / 2` 
Because bottom position will be calculated by adding this value to the **center** of the view not the top of the view.

MFlisar replied:
done, did not see this mistake...

faruktoptas commented on app/build.gradle:
```diff
@@ -52,12 +50,12 @@ android {
     implementation project(':fancyshowcaseview')
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
-    androidTestImplementation 'com.android.support.test.espresso:espresso-core:2.2.2', {
+    androidTestImplementation 'androidx.test.espresso:espresso-core:3.1.0', {
```
Missing `androidTestImplementation 'androidx.test:rules:1.1.1'`

MFlisar replied:
added

faruktoptas commented:
Thank you @MFlisar :)
Can you check that instrumentation tests are not broken by running `./gradlew connectedAndroidTest`

MFlisar commented:
Added following to the demo app:

    androidTestImplementation 'androidx.test.ext:junit:1.1.0'
    androidTestImplementation 'androidx.test:rules:1.1.1'

And fixed the error I made in the calculator.

Finally I tested `connectedAndroidTest` and it's running successfully now.

MFlisar commented:
Any plans on merging this PR? Is there still something missing for you?

faruktoptas commented:
@MFlisar  Sorry for the delay. I will merge it asap.

faruktoptas commented:
@MFlisar finally merged! Sorry for delayed response :)
I will revert Material style to AppCompat.

MFlisar commented:
Any good reason for this? There are bridge themes to fill the gap between material components and app compat... so no one needs to migrate to material components theme if they don't want to, they can use the bridge themes and all app compat styles will continue to work...

faruktoptas commented:
I changed the theme for sample app only not for the library.
